### PR TITLE
Bug 1421303 - New XCUITests: New Tab Settings

### PR DESCRIPTION
**Pull request by isabelrios**
The aim of this PR is to add tests to verify the New Tab Settings option in settings:

Settings -> New Tab -> New Tab: Show Blank Page, Show your Top Sites (default), Show your Bookmarks, Show your History

### CODE REVIEW

File: XCUITests/NewTabSettings.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+        //This appears under top sites
+        waitforNoExistence(app.staticTexts["Highlights"])
+
+        // Enable toggle again and check it is shown
+        navigator.goto(NewTabSettings)
+        navigator.toggleOn(userState.bookmarksInNewTab, withAction: Action.ToggleBookmarksInNewTab)
+        navigator.goto(HomePanelsScreen)
+        navigator.goto(NewTabScreen)
+        navigator.nowAt(NewTabScreen)
```
Comment (jhugman) [inline XCUITests/NewTabSettings.swift]: This doesn't make sense (`goto` followed by a `nowAt`)

File: XCUITests/NewTabSettings.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        waitforNoExistence(app.staticTexts["Highlights"])
+    }
+
+    func testChangeNewTabSettingsShowYourBookmarks() {
+        navigator.goto(NewTabChoiceSettings)
+        waitforExistence(app.tables["NewTabPage.Setting.Options"])
+        // Show Bookmarks panel without bookmarks
+        navigator.performAction(Action.ShowBookmarksPage)
+        navigator.goto(HomePanelsScreen)
+        navigator.goto(NewTabScreen)
```
Comment (jhugman) [inline XCUITests/NewTabSettings.swift]: You do this several times in this test: goto `HomePanelsScreen` then `NewTabScreen`. Why?

File: XCUITests/FxScreenGraph.swift [hunk 3]
```diff
@@ -143,6 +143,12 @@ class Action {
     static let DisablePasscodeTypeIncorrectPasscode = "DisablePasscodeTypeIncorrectPasscode"
 
     static let TogglePocketInNewTab = "TogglePocketInNewTab"
+    static let ToggleBookmarksInNewTab = "ToggleBookmarksInNewTab"
+    static let ToggleHistoryInNewTab = "ToggleHistoryInNewTab"
+
+    static let ShowBlankPage = "ShowBlankPage"
+    static let ShowBookmarksPage = "ShowBookmarksPage"
+    static let ShowHistoryPage = "ShowHistoryPage"
```
Comment (jhugman) [inline XCUITests/FxScreenGraph.swift]: I think these should be named as `SelectNewTabAsBlankPage` or something to suggest that they're Settings actions.

Comment (isabelrios) [reply XCUITests/FxScreenGraph.swift]: 👍 Agree, will change the name

File: XCUITests/FxScreenGraph.swift [hunk 4]
```diff
@@ -468,12 +476,29 @@ func createScreenGraph(for test: XCTestCase, with app: XCUIApplication) -> Scree
...
+        }
         screenState.backAction = navigationControllerBackAction
     }
 
     map.addScreenState(NewTabChoiceSettings) { screenState in
-        let table = app.tables.element(boundBy: 0)
+        let table = app.tables["NewTabPage.Setting.Options"]
         screenState.backAction = navigationControllerBackAction
+        screenState.gesture(forAction: Action.ShowBlankPage) { UserState in
+            table.cells.staticTexts["Show a Blank Page"].tap()
```
[the remark this answers is not in the document]
Comment (isabelrios) [reply XCUITests/FxScreenGraph.swift]: These accessibilityIdentifier are not defined. Will try to create them

File: XCUITests/NewTabSettings.swift (hunk 1 above)
Comment (isabelrios) [reply XCUITests/NewTabSettings.swift]: You are right, I removed that line

File: XCUITests/NewTabSettings.swift (hunk 2 above)
Comment (isabelrios) [reply XCUITests/NewTabSettings.swift]: I needed to open a new tab to check that the setting selected was working that's whay I did it that way. Just realized I can change those two lines with: ```navigator.performAction(Action.OpenNewTabFromTabTray)```
I will update the tests.
Thanks

File: XCUITests/FxScreenGraph.swift [hunk 5]
```diff
@@ -146,9 +146,9 @@ class Action {
     static let ToggleBookmarksInNewTab = "ToggleBookmarksInNewTab"
     static let ToggleHistoryInNewTab = "ToggleHistoryInNewTab"
 
-    static let ShowBlankPage = "ShowBlankPage"
-    static let ShowBookmarksPage = "ShowBookmarksPage"
-    static let ShowHistoryPage = "ShowHistoryPage"
+    static let SelectNewTabAsBlankPage = "SelectNewTabAsBlankPage"
+    static let SelectNewTabAsBookmarksPage = "SelectNewTabAsBookmarksPage"
+    static let SelectNewTabAsHistoryPage = "SelectNewTabAsHistoryPage"
```
Comment (jhugman) [inline XCUITests/FxScreenGraph.swift]: Good.

File: XCUITests/FxScreenGraph.swift [hunk 6]
```diff
@@ -490,14 +490,14 @@ func createScreenGraph(for test: XCTestCase, with app: XCUIApplication) -> Scree
...
-        screenState.gesture(forAction: Action.ShowBookmarksPage) { UserState in
-            table.cells.staticTexts["Show your Bookmarks"].tap()
+        screenState.gesture(forAction: Action.SelectNewTabAsBookmarksPage) { UserState in
+            table.cells["Bookmarks"].tap()
         }
-        screenState.gesture(forAction: Action.ShowHistoryPage) { UserState in
-            table.cells.staticTexts["Show your History"].tap()
+        screenState.gesture(forAction: Action.SelectNewTabAsHistoryPage) { UserState in
+            table.cells["History"].tap()
         }
```
Comment (jhugman) [inline XCUITests/FxScreenGraph.swift]: Yep.

File: XCUITests/NewTabSettings.swift [hunk 7]
```diff
@@ -70,9 +65,8 @@ class NewTabSettingsTest: BaseTestCase {
         navigator.goto(NewTabChoiceSettings)
         waitforExistence(app.tables["NewTabPage.Setting.Options"])
         // Show History Panel without history
-        navigator.performAction(Action.ShowHistoryPage)
-        navigator.goto(HomePanelsScreen)
-        navigator.goto(NewTabScreen)
+        navigator.performAction(Action.SelectNewTabAsHistoryPage)
+        navigator.performAction(Action.OpenNewTabFromTabTray)
```
Comment (jhugman) [inline XCUITests/NewTabSettings.swift]: Excellent. So much clearer.

File: Client/Frontend/Settings/NewTabChoiceViewController.swift [hunk 8]
```diff
@@ -69,7 +69,7 @@ class NewTabChoiceViewController: UITableViewController {
         cell.accessoryType = (currentChoice == option) ? .checkmark : .none
         cell.textLabel?.attributedText = NSAttributedString.tableRowTitle(option.settingTitle, enabled: enabled)
         cell.isUserInteractionEnabled = enabled
-
+        cell.accessibilityIdentifier = option.rawValue
```
Comment (jhugman) [inline Client/Frontend/Settings/NewTabChoiceViewController.swift]: Good.

**Pull request merged**